kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java

```diff
@@ -613,6 +616,19 @@ public String toString() {
     }
   }
 
+  private static class BucketNameValidator implements ConfigDef.Validator {
+    @Override
+    public void ensureValid(String name, Object bucket) {
+      String bucketName = ((String) bucket).trim();
+      try {
+        BucketNameUtils.validateBucketName(bucketName);
+      } catch (IllegalArgumentException e) {
+        throw new ConfigException(e.getMessage() + ". " + bucketName
+                + " is an invalid bucket name. Does not follow AWS guidelines.");
+      }
+    }
```

Comment:
We should always have a `toString()` method on all validators that's used in our documentation, and the string should describe the limitation. For this, something like:
> Follows the AWS guidelines for S3 bucket names